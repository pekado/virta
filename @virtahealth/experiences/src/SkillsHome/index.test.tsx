import * as React from "react";
import { render, waitFor } from "@testing-library/react-native";
import { testClient, WrapComponentWithVirtaContext } from "../test";
import { SkillsHome } from "./index";

describe("Skills Home", () => {
  it("renders", async () => {
    testClient.get.mockReturnValueOnce(Promise.resolve(mockSkillActions));
    const skillsHome = render(
      <WrapComponentWithVirtaContext>
        <SkillsHome virtaId={"123abc"} />
      </WrapComponentWithVirtaContext>
    );
    await waitFor(() => expect(testClient.get).toBeCalled());

    skillsHome.update(
      <WrapComponentWithVirtaContext>
        <SkillsHome virtaId={"123abc"} />
      </WrapComponentWithVirtaContext>
    );

    // TODO - this should work but doesn't due to a bug
    // https://github.com/callstack/react-native-testing-library/issues/553
    // once the bug is fixed, switch back to the original test
    // waitFor(() => {
    //   skillsHome.queryByText("Meal Planning Basics");
    // });
    expect(
      skillsHome.toJSON().children[1].children[0].children[0].children[0]
        .children[0].children[0]
    ).toEqual("Meal Planning Basics");
    expect(skillsHome).toMatchSnapshot();
  });
});

const mockSkillActions = [
  {
    id: 1,
    title: "Meal Planning Basics",
    url: "https://wikipedia.org",
  },
  {
    id: 2,
    title: "Balancing Electrolytes",
    url: "https://my.virtahealth.com",
  },
  {
    id: 3,
    title: "WOOP My Life",
    url: "https://my.virtahealth.com",
  },
  {
    id: 4,
    title: "Emotional Agility",
    url: "https://my.virtahealth.com",
  },
  {
    id: 5,
    title: "Finding the Right Fats",
    url: "https://my.virtahealth.com",
  },
  {
    id: 6,
    title: "Finding the Right Fats ",
    url: "https://my.virtahealth.com",
  },
  {
    id: 7,
    title: "Finding the Right Fats  ",
    url: "https://my.virtahealth.com",
  },
  {
    id: 8,
    title: "Finding the Right Fats   ",
    url: "https://my.virtahealth.com",
  },
  {
    id: 9,
    title: "Finding the Right Fat",
    url: "https://my.virtahealth.com",
  },
];
