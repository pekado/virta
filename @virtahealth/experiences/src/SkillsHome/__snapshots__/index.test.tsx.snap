// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills Home renders 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(58, 61, 67)",
          "fontFamily": "Whitney-Book",
          "fontSize": 18,
          "fontWeight": "300",
          "marginBottom": 10,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 10,
        },
      ]
    }
  >
    Build your VirtaLife skills to maintain or level up your progress.
  </Text>
  <View
    screenWidth={750}
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      screenWidth={750}
      style={
        Array [
          Object {
            "height": 72,
            "marginBottom": 7,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 7,
            "width": "32%",
          },
        ]
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Array [
            Object {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          backgroundColor="linkTilePurpleBackground"
          style={
            Array [
              Object {
                "backgroundColor": "rgb(193, 158, 212)",
                "borderRadius": 10,
                "display": "flex",
                "height": "100%",
                "justifyContent": "flex-start",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <Text
            labelColor="linkTilePurpleText"
            style={
              Array [
                Object {
                  "color": "rgb(63, 47, 120)",
                  "fontFamily": "Whitney-Semibold",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "marginBottom": 10,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 10,
                },
              ]
            }
          >
            Meal Planning Basics
          </Text>
        </View>
      </TouchableOpacity>
    </View>
    <View
      screenWidth={750}
      style={
        Array [
          Object {
            "height": 72,
            "marginBottom": 7,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 7,
            "width": "32%",
          },
        ]
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Array [
            Object {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          backgroundColor="linkTileOrangeBackground"
          style={
            Array [
              Object {
                "backgroundColor": "rgb(252, 176, 65)",
                "borderRadius": 10,
                "display": "flex",
                "height": "100%",
                "justifyContent": "flex-start",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <Text
            labelColor="linkTileOrangeText"
            style={
              Array [
                Object {
                  "color": "rgb(127, 58, 5)",
                  "fontFamily": "Whitney-Semibold",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "marginBottom": 10,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 10,
                },
              ]
            }
          >
            Balancing Electrolytes
          </Text>
        </View>
      </TouchableOpacity>
    </View>
    <View
      screenWidth={750}
      style={
        Array [
          Object {
            "height": 72,
            "marginBottom": 7,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 7,
            "width": "32%",
          },
        ]
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Array [
            Object {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          backgroundColor="linkTileBlueBackground"
          style={
            Array [
              Object {
                "backgroundColor": "rgb(158, 219, 255)",
                "borderRadius": 10,
                "display": "flex",
                "height": "100%",
                "justifyContent": "flex-start",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <Text
            labelColor="linkTileBlueText"
            style={
              Array [
                Object {
                  "color": "rgb(13, 70, 129)",
                  "fontFamily": "Whitney-Semibold",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "marginBottom": 10,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 10,
                },
              ]
            }
          >
            WOOP My Life
          </Text>
        </View>
      </TouchableOpacity>
    </View>
    <View
      screenWidth={750}
      style={
        Array [
          Object {
            "height": 72,
            "marginBottom": 7,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 7,
            "width": "32%",
          },
        ]
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Array [
            Object {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          backgroundColor="linkTileRedBackground"
          style={
            Array [
              Object {
                "backgroundColor": "rgb(252, 142, 135)",
                "borderRadius": 10,
                "display": "flex",
                "height": "100%",
                "justifyContent": "flex-start",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <Text
            labelColor="linkTileRedText"
            style={
              Array [
                Object {
                  "color": "rgb(134, 9, 11)",
                  "fontFamily": "Whitney-Semibold",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "marginBottom": 10,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 10,
                },
              ]
            }
          >
            Emotional Agility
          </Text>
        </View>
      </TouchableOpacity>
    </View>
    <View
      screenWidth={750}
      style={
        Array [
          Object {
            "height": 72,
            "marginBottom": 7,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 7,
            "width": "32%",
          },
        ]
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Array [
            Object {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          backgroundColor="linkTileOrangeBackground"
          style={
            Array [
              Object {
                "backgroundColor": "rgb(252, 176, 65)",
                "borderRadius": 10,
                "display": "flex",
                "height": "100%",
                "justifyContent": "flex-start",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <Text
            labelColor="linkTileOrangeText"
            style={
              Array [
                Object {
                  "color": "rgb(127, 58, 5)",
                  "fontFamily": "Whitney-Semibold",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "marginBottom": 10,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 10,
                },
              ]
            }
          >
            Finding the Right Fats
          </Text>
        </View>
      </TouchableOpacity>
    </View>
    <View
      screenWidth={750}
      style={
        Array [
          Object {
            "height": 72,
            "marginBottom": 7,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 7,
            "width": "32%",
          },
        ]
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Array [
            Object {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          backgroundColor="linkTileRedBackground"
          style={
            Array [
              Object {
                "backgroundColor": "rgb(252, 142, 135)",
                "borderRadius": 10,
                "display": "flex",
                "height": "100%",
                "justifyContent": "flex-start",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <Text
            labelColor="linkTileRedText"
            style={
              Array [
                Object {
                  "color": "rgb(134, 9, 11)",
                  "fontFamily": "Whitney-Semibold",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "marginBottom": 10,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 10,
                },
              ]
            }
          >
            Finding the Right Fats 
          </Text>
        </View>
      </TouchableOpacity>
    </View>
    <View
      screenWidth={750}
      style={
        Array [
          Object {
            "height": 72,
            "marginBottom": 7,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 7,
            "width": "32%",
          },
        ]
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Array [
            Object {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          backgroundColor="linkTileBlueBackground"
          style={
            Array [
              Object {
                "backgroundColor": "rgb(158, 219, 255)",
                "borderRadius": 10,
                "display": "flex",
                "height": "100%",
                "justifyContent": "flex-start",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <Text
            labelColor="linkTileBlueText"
            style={
              Array [
                Object {
                  "color": "rgb(13, 70, 129)",
                  "fontFamily": "Whitney-Semibold",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "marginBottom": 10,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 10,
                },
              ]
            }
          >
            Finding the Right Fats  
          </Text>
        </View>
      </TouchableOpacity>
    </View>
    <View
      screenWidth={750}
      style={
        Array [
          Object {
            "height": 72,
            "marginBottom": 7,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 7,
            "width": "32%",
          },
        ]
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Array [
            Object {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          backgroundColor="linkTileGreenBackground"
          style={
            Array [
              Object {
                "backgroundColor": "rgb(112, 200, 181)",
                "borderRadius": 10,
                "display": "flex",
                "height": "100%",
                "justifyContent": "flex-start",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <Text
            labelColor="linkTileGreenText"
            style={
              Array [
                Object {
                  "color": "rgb(0, 82, 52)",
                  "fontFamily": "Whitney-Semibold",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "marginBottom": 10,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 10,
                },
              ]
            }
          >
            Finding the Right Fats   
          </Text>
        </View>
      </TouchableOpacity>
    </View>
    <View
      screenWidth={750}
      style={
        Array [
          Object {
            "height": 72,
            "marginBottom": 7,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 7,
            "width": "32%",
          },
        ]
      }
    >
      <TouchableOpacity
        onPress={[Function]}
        style={
          Array [
            Object {
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <View
          backgroundColor="linkTilePinkBackground"
          style={
            Array [
              Object {
                "backgroundColor": "rgb(252, 156, 202)",
                "borderRadius": 10,
                "display": "flex",
                "height": "100%",
                "justifyContent": "flex-start",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          <Text
            labelColor="linkTilePinkText"
            style={
              Array [
                Object {
                  "color": "rgb(143, 1, 99)",
                  "fontFamily": "Whitney-Semibold",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "marginBottom": 10,
                  "marginLeft": 10,
                  "marginRight": 10,
                  "marginTop": 10,
                },
              ]
            }
          >
            Finding the Right Fat
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
